platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,426 @@
+        """
+        Fetches user-related data, including user ID, email, and name.
+
+        Returns:
+            dict: A dictionary containing user ID, email, and name.
+        """
+        user_service = self.runtime.service(self, "user")
+        current_user = user_service.get_current_user()
+        return {
+            "user_id": current_user.opt_attrs["edx-platform.user_id"],
+            "user_email": current_user.emails[0],
+            "user_name": current_user.full_name.split(),
+        }
+
+    @XBlock.json_handler
+    def get_eula_agreement(self, data, suffix=""):
+        """
+        Fetches the End User License Agreement (EULA) content.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: A dictionary containing the HTML content of the EULA and the status code.
+        """
+        response = get_eula_page()
+        return {"html": response.text, "status": response.status_code}
+
+    @XBlock.json_handler
+    def accept_eula_agreement(self, data, suffix=""):
+        """
+        Submits acceptance of the EULA for the current user.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: The response after accepting the EULA.
+        """
+        user_id = self.get_user_data()["user_id"]
+        date_now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+        payload = {
+            "user_id": str(user_id),
+            "accepted_timestamp": date_now,
+            "language": "en-US",
+        }
+        response = post_accept_eula_version(payload)
+        return response.json()
+
+    def create_turnitin_submission_object(self):
+        """
+        Constructs a Turnitin submission object based on the user's data.
+
+        Returns:
+            Response: The response from the Turnitin submission API.
+        """
+        user_data = self.get_user_data()
+        user_name = user_data["user_name"]
+        date_now = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+        payload = {
+            "owner": user_data["user_id"],
+            "title": self.location.block_id,
+            "submitter": user_data["user_id"],
+            "owner_default_permission_set": "LEARNER",
+            "submitter_default_permission_set": "INSTRUCTOR",
+            "extract_text_only": False,
+            "metadata": {
+                "owners": [
+                    {
+                        "id": user_data["user_id"],
+                        "given_name": user_name[0] if user_name else "no_name",
+                        "family_name": " ".join(user_name[1:])
+                        if len(user_name) > 1
+                        else "no_last_name",
+                        "email": user_data["user_email"],
+                    }
```

Comment:
I think you can get the `firstname` and `lastname` in `get_user_data` instead of here, so the logic is more encapsulated